File: day17_part1_2015.rs

use std::fs;

fn main() {
    let input = fs::read_to_string("input.txt").unwrap();
    let containers: Vec<i32> = input.lines().map(|x| x.parse().unwrap()).collect();
    
    let mut count = 0;
    for i in 1..(1 << containers.len()) {
        let mut sum = 0;
        for j in 0..containers.len() {
            if (i >> j) & 1 == 1 {
                sum += containers[j];
            }
        }
        if sum == 150 {
            count += 1;
        }
    }
    
    println!("{}", count);
}

File: objectivec/day17_part1_2015.m
#import <Foundation/Foundation.h>

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        NSString *input = [NSString stringWithContentsOfFile:@"input.txt" encoding:NSUTF8StringEncoding error:nil];
        NSArray *containers = [input componentsSeparatedByString:@"\n"];
        
        NSMutableArray *containerSizes = [NSMutableArray array];
        for (NSString *container in containers) {
            [containerSizes addObject:@(container.intValue)];
        }
        
        int target = 150;
        int count = 0;
        
        for (int i = 1; i < (1 << containerSizes.count); i++) {
            int sum = 0;
            for (int j = 0; j < containerSizes.count; j++) {
                if (i & (1 << j)) {
                    sum += [containerSizes[j] intValue];
                }
            }
            if (sum == target) {
                count++;
            }
        }
        
        printf("%d\n", count);
    }
    return 0;
}